### ISSUE TITLE
False positive with Style/CombinableLoops and Hash access with safe operator 

### ISSUE THREAD

**Issue by JasonBarnabe** [rubocop/rubocop#10319]
`Style/Combinable` loops has a false positive when dealing with consecutive loops that based on a Hash using the safe operator.

```sh
rubocop --only Style/CombinableLoops --debug rubocoptest.rb
```

```ruby
params = {}

params[:a]&.each do |v|
  puts v
end
params[:b]&.each do |v|
  puts v
end
```

Dropping the `&`s makes it not flag this.

--------

## Expected behavior

Not an offense.

## Actual behavior

```
For /www/greasyfork: configuration from /www/greasyfork/.rubocop.yml
configuration from /home/jason/.rbenv/versions/3.0.3/lib/ruby/gems/3.0.0/gems/rubocop-rails-2.13.0/config/default.yml
configuration from /home/jason/.rbenv/versions/3.0.3/lib/ruby/gems/3.0.0/gems/rubocop-rails-2.13.0/config/default.yml
Default configuration from /home/jason/.rbenv/versions/3.0.3/lib/ruby/gems/3.0.0/gems/rubocop-1.24.0/config/default.yml
configuration from /home/jason/.rbenv/versions/3.0.3/lib/ruby/gems/3.0.0/gems/rubocop-minitest-0.17.0/config/default.yml
configuration from /home/jason/.rbenv/versions/3.0.3/lib/ruby/gems/3.0.0/gems/rubocop-minitest-0.17.0/config/default.yml
configuration from /home/jason/.rbenv/versions/3.0.3/lib/ruby/gems/3.0.0/gems/rubocop-performance-1.13.0/config/default.yml
configuration from /home/jason/.rbenv/versions/3.0.3/lib/ruby/gems/3.0.0/gems/rubocop-performance-1.13.0/config/default.yml
Use parallel by default.
Running parallel inspection
Loading cache from /home/jason/.cache/rubocop_cache/8ef243d90315076c271fe7c82988bab78314d142/31d90c7c0e4385ebe1fe03d808cc1a706ae37e55/e8a486d5299c4f4beb2851aabb2371148602e0c0
Inspecting 1 file
Scanning /www/greasyfork/rubocoptest.rb
Loading cache from /home/jason/.cache/rubocop_cache/8ef243d90315076c271fe7c82988bab78314d142/31d90c7c0e4385ebe1fe03d808cc1a706ae37e55/e8a486d5299c4f4beb2851aabb2371148602e0c0
C

Offenses:

rubocoptest.rb:6:1: C: Style/CombinableLoops: Combine this loop with the previous loop.
params[:b]&.each do |v| ...
^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
Finished in 0.38021430297521874 seconds
```

## RuboCop version

```
1.24.0 (using Parser 3.0.3.2, rubocop-ast 1.15.0, running on ruby 3.0.3 x86_64-linux)
  - rubocop-minitest 0.17.0
  - rubocop-performance 1.13.0
  - rubocop-rails 2.13.0
```
